token/js/test/e2e-2022/tlv.test.ts

```diff
@@ -0,0 +1,100 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import type { Connection, PublicKey, Signer } from '@solana/web3.js';
+import { sendAndConfirmTransaction, Keypair, SystemProgram, Transaction } from '@solana/web3.js';
+
+import type { Account, Mint } from '../../src';
+import {
+    createInitializeAccountInstruction,
+    getAccount,
+    getAccountLen,
+    createMint,
+    ExtensionType,
+    getExtensionData,
+} from '../../src';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+
+const TEST_TOKEN_DECIMALS = 9;
+const ACCOUNT_EXTENSIONS = [
+    ExtensionType.TransferFeeAmount,
+    ExtensionType.ConfidentialTransferAccount,
+    ExtensionType.ImmutableOwner,
+    ExtensionType.MemoTransfer,
+    ExtensionType.CpiGuard,
+];
```

Comment:
nit take-it-or-leave-it: wanna add this to `extension/extensionType.ts`? Otherwise I don't think we'll remember to update this whenever we add a new account extension type. on the flip side, if we forget here, I'm not 100% sure we'll remember to update it elsewhere

Reply:
i ended up adding usable predicates to `extensionType.ts` and a unit test for them instead